### DISCUSSION TITLE
How does the ping/pong work behind the scenes?

### DISCUSSION THREAD

**Discussion by crashedpixel** [uNetworking/uWebSockets.js#533] [Q&A]
Trello for example you can see all of the ping/ pongs events in the browser under the network tab for that Websocket connection.

Whereas I can see the pong events on the server for uWebSocket but I can't see any network activity for the ping events in the browser. I know they're "automatic" but I would see expect to see something in the network tab.

Is this new protocol for Websockets? I tried looking through the source code for uWebSocket but not so good with C++.

**Comment by ghost** [comment]
Pings are not shown in the browser. It's nothing new -> the browser sends pongs for every ping (under the hood).

**Reply by ghost** [reply]
You can use WireShark for proof of this
